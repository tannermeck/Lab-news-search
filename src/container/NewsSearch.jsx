import React, { Component } from 'react';
import { ArticleList } from '../presentation/ArticleList';
import { SearchArticles } from '../presentation/SearchArticles';
import { fetchAnyArticle } from '../utils/fetchAnyArticle';
import fetchApi from '../utils/fetchApi';

class NewsSearch extends Component {
    state = { 
        loading: true,
        articles: [],
        search: ''
    }

    componentDidMount = async () => {
        const articles = await fetchApi();
        this.setState({ articles, loading: false })
    }

    handleSearchChange = ({target}) => {
        this.setState({ search: target.value })
    }
    onSearchSubmit = async(event) => {
        event.preventDefault();
        this.setState({loading: true})
        const articles = await fetchAnyArticle(this.state.search);
        this.setState({articles: articles, loading: false})
    }

    render(){
        console.log('articles', this.state.articles)
        const { articles, search, loading } = this.state
        if (loading) return <h1>Loading...</h1>
        return (
            <>
                <SearchArticles onSubmit={this.onSearchSubmit} query={search} handleChange={this.handleSearchChange}/>
                <ArticleList articles={articles}/>
            </>
        )
    }
}

export default NewsSearch;