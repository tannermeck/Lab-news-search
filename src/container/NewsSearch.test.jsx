import React from 'react';
import userEvent from '@testing-library/user-event';
import { render, screen, waitFor } from '@testing-library/react';
import NewsSearch from './NewsSearch';

describe('shows a list of articles', () => {
    it('should display a list of authors, titles, and description', async () => {
        render(<NewsSearch />)
        console.log('API_KEY', process.env.REACT_APP_API_KEY)
        screen.getByText('Loading...');
        const ul = await screen.findByRole('list', { name: 'articles' });
        expect(ul).toMatchSnapshot();
        const input = await screen.findByLabelText('Find Articles')
        userEvent.type(input, 'health');
        const submit = await screen.findByRole('button', { name: 'button' })
        userEvent.click(submit);
        return waitFor(() => {
            const quotes = screen.getAllByText('Description', 
            {exact: false})
            expect(quotes).toHaveLength(20);
        })
    })
}) 