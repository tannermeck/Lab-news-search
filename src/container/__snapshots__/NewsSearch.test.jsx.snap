// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows a list of articles should display a list of authors, titles, and description 1`] = `
<ul
  aria-label="articles"
>
  <li>
    <p
      aria-label="author-tag"
    >
      Author: 
      noreply@blogger.com (Unknown)
    </p>
    <p>
      Title: 
      Bitcoin hits strongest level since May
    </p>
    <p>
      Description: 
      &lt;ul&gt;&lt;li&gt;Bitcoin, in terms of market value, rose 4.6% to $53,859.6.&lt;/li&gt;&lt;li&gt;It passed $50,000 mark for first time in four weeks on Tuesday. &lt;/li&gt;&lt;li&gt;Bitcoin fell below $50,000 in early September.&lt;/li&gt;&lt;/ul&gt;Bitcoin hit its highest level since mid-May on Wednesda…
    </p>
  </li>
  <li>
    <p
      aria-label="author-tag"
    >
      Author: 
      noreply@blogger.com (Unknown)
    </p>
    <p>
      Title: 
      JPMorgan CEO Says Bitcoin Has No Intrinsic Value After Claiming That Its Price Could Rise 10X
    </p>
    <p>
      Description: 
      JPMorgan CEO Jamie Dimon is still not a Bitcoin fan despite rapid price appreciation   from U.Today https://ift.tt/3iyi5nR https://ift.tt/eA...
    </p>
  </li>
  <li>
    <p
      aria-label="author-tag"
    >
      Author: 
      noreply@blogger.com (Unknown)
    </p>
    <p>
      Title: 
      BTC, DOGE, and SHIBA Price Analysis for October 6
    </p>
    <p>
      Description: 
      How high are the chances of Bitcoin sustaining its gains and pushing altcoins higher?   from U.Today https://ift.tt/3BkD5px https://ift.tt/e...
    </p>
  </li>
  <li>
    <p
      aria-label="author-tag"
    >
      Author: 
      noreply@blogger.com (Unknown)
    </p>
    <p>
      Title: 
      Who Bought $1.6B in Bitcoin Wednesday, and Why?
    </p>
    <p>
      Description: 
      last week the cryptocurrency market persistently asked the gnawing and annoying question, “Why?”
Specifically, why did someone make a massive purchase of $1.6 billion worth of bitcoin on Wednesday in a couple of minutes?
While many see this huge buy as a si…
    </p>
  </li>
  <li>
    <p
      aria-label="author-tag"
    >
      Author: 
      Molly Taft
    </p>
    <p>
      Title: 
      Bitcoin Miners Are Gobbling Up U.S. Energy
    </p>
    <p>
      Description: 
      There’s a big new presence slurping up power from the U.S. grid, and it’s growing: bitcoin miners. New research shows that the U.S. has overtaken China as the top global destination for bitcoin mining and energy use is skyrocketing as a result.Read more...
    </p>
  </li>
  <li>
    <p
      aria-label="author-tag"
    >
      Author: 
      Molly Taft
    </p>
    <p>
      Title: 
      Mining Bitcoin Using Nuclear Power May Be Fine, Actually
    </p>
    <p>
      Description: 
      Last week, the Wall Street Journal ran a piece on three recent nuclear-bitcoin deals that may signal a growing trend in the industry. The Journal piece reflects a small but growing sense of excitement expressed by some crypto fans—including Miami mayor and bi…
    </p>
  </li>
  <li>
    <p
      aria-label="author-tag"
    >
      Author: 
      Tom McKay
    </p>
    <p>
      Title: 
      Roughly One-Third of Bitcoin Is Controlled by a Small Cabal of Whales, According to New Study
    </p>
    <p>
      Description: 
      For all the talk of democratizing finance, the vast majority of Bitcoin continues to be owned by a relative handful of investors.Read more...
    </p>
  </li>
  <li>
    <p
      aria-label="author-tag"
    >
      Author: 
      noreply@blogger.com (Unknown)
    </p>
    <p>
      Title: 
      SEC Chair Dodges Question About Regulatory Status of Bitcoin and Ether
    </p>
    <p>
      Description: 
      The SEC boss remains mum about the regulatory status of the two largest cryptocurrenciesfrom U.Today https://ift.tt/3Af9H2O https://ift.tt/eA8V8J
    </p>
  </li>
  <li>
    <p
      aria-label="author-tag"
    >
      Author: 
      noreply@blogger.com (Unknown)
    </p>
    <p>
      Title: 
      IMF Head Says Central Bank Digital Currencies Are Reliable, Hard to Think of Bitcoin as Money
    </p>
    <p>
      Description: 
      Digital currencies backed by central banks are the most reliable form of digital money, according to IMF managing director Kristalina Georgieva. At the same time, to think of cryptocurrencies such as bitcoin as money is difficult, believes the chief executive…
    </p>
  </li>
  <li>
    <p
      aria-label="author-tag"
    >
      Author: 
      Whitney Kimball
    </p>
    <p>
      Title: 
      A Bitcoin Mining Operation Started a Secret Power Plant, It Did Not Go as Planned
    </p>
    <p>
      Description: 
      A There Will Be Blood-style drama unfolds in Canada, where a Bitcoin mining operation illegally tapped an unused natural gas well near a wealthy neighborhood in Alberta. The company Link Global might have gotten away with pilfering gas, if not for noise compl…
    </p>
  </li>
  <li>
    <p
      aria-label="author-tag"
    >
      Author: 
    </p>
    <p>
      Title: 
      Bitcoin edges off all-time high but momentum for more gains this year seen intact - Reuters
    </p>
    <p>
      Description: 
      Bitcoin fell slightly in Asian hours on Thursday, a day after marking an all-time high on optimism around the launch of the first U.S. bitcoin futures ETF.
    </p>
  </li>
  <li>
    <p
      aria-label="author-tag"
    >
      Author: 
    </p>
    <p>
      Title: 
      Riding the crypto rollercoaster: Bitcoin nears record high - Reuters
    </p>
    <p>
      Description: 
      Bitcoin is on the cusp of an all-time high, its latest rally fuelled by the launch of the first U.S. bitcoin futures exchange-traded fund that investors think could make the cryptocurrency accessible for swathes of new investors. &lt;a href="/technology/bitcoin-…
    </p>
  </li>
  <li>
    <p
      aria-label="author-tag"
    >
      Author: 
      ilee@insider.com (Isabelle Lee)
    </p>
    <p>
      Title: 
      The SEC could approve as many as 4 bitcoin futures ETFs in October
    </p>
    <p>
      Description: 
      Unlike bitcoin ETFs, SEC Chair Gary Gensler has expressed more openness towards bitcoin futures ETF.
    </p>
  </li>
  <li>
    <p
      aria-label="author-tag"
    >
      Author: 
    </p>
    <p>
      Title: 
      Bitcoin nears record high ahead of futures ETF listing - Reuters
    </p>
    <p>
      Description: 
      Bitcoin hit a six-month high and was within striking distance of a record on Tuesday as traders bet an anticipated listing of a futures-based U.S. exchange traded fund could herald investment flows into bitcoin and cryptocurrency assets.
    </p>
  </li>
  <li>
    <p
      aria-label="author-tag"
    >
      Author: 
    </p>
    <p>
      Title: 
      Bitcoin sits below all-time high after U.S. ETF debut - Reuters
    </p>
    <p>
      Description: 
      Bitcoin  hovered below record levels on Wednesday, the day after the first U.S. bitcoin futures-based exchange-traded fund (ETF) began trading, a development that market participants say is likely to drive investment into the digital asset.
    </p>
  </li>
  <li>
    <p
      aria-label="author-tag"
    >
      Author: 
      Reuters Editorial
    </p>
    <p>
      Title: 
      Bitcoin tops $67,000 in record high - Reuters.com
    </p>
    <p>
      Description: 
      Bitcoin climbed to a record high on Wednesday amid new interest as the first U.S. bitcoin futures-based exchange-traded funds come to market.  Conway G. Gittens has more.
    </p>
  </li>
  <li>
    <p
      aria-label="author-tag"
    >
      Author: 
      Reuters Staff
    </p>
    <p>
      Title: 
      Bitcoin nears record high ahead of futures ETF listing - Reuters
    </p>
    <p>
      Description: 
      Bitcoin hit a six-month high and was within striking distance of a record on Tuesday as traders bet an anticipated listing of a futures-based U.S. exchange traded fund could herald investment flows into bitcoin and cryptocurrency assets.
    </p>
  </li>
  <li>
    <p
      aria-label="author-tag"
    >
      Author: 
      insider@insider.com (Carla Mozée)
    </p>
    <p>
      Title: 
      The first-ever bitcoin ETF will start trading on Tuesday
    </p>
    <p>
      Description: 
      The arrival of the bitcoin-futures ETF comes as bitcoin this year has moved above $1 trillion in market capitalization.
    </p>
  </li>
  <li>
    <p
      aria-label="author-tag"
    >
      Author: 
      insider@insider.com (Carla Mozée)
    </p>
    <p>
      Title: 
      SEC punts on approving a spot bitcoin ETF again, pushing decision on Valkyrie's fund to 2022
    </p>
    <p>
      Description: 
      While investors wait for word on a spot bitcoin ETF, the SEC has let bitcoin futures ETFs launch.
    </p>
  </li>
  <li>
    <p
      aria-label="author-tag"
    >
      Author: 
      Gertrude Chavez-Dreyfuss
    </p>
    <p>
      Title: 
      ANALYSIS-Bitcoin futures highlight some pitfalls for new ETFs - Reuters
    </p>
    <p>
      Description: 
      ANALYSIS-Bitcoin futures highlight some pitfalls for new ETFs  Reuters
    </p>
  </li>
</ul>
`;
